### DISCUSSION TITLE
How to use query push using useRouter in NextJS 13 App Router?

### DISCUSSION THREAD

**Discussion by abhiishekbhatta** [vercel/next.js#51983] [Help]
### Summary

In new App router, I tried this

`<button onClick={() => {
   router.push('/donate/payment?amount=20')
 }}>Donate</button>
`
and this

`<button onClick={() => {
   router.push({
      pathname: '/donate/payment',
      query: { amount:20 }
    })
 }}>Donate</button>
`
but neither of this work.

**Comment by icyJoseph** [comment]
Hi,

The first one should've worked. 

Are you taking `useRouter` from `next/navigation`?

**Reply by abhiishekbhatta** [reply]
yes I am using `useRouter` from `next/navigation`

**Reply by icyJoseph** [reply]
And you used the `'use client'`directive? I am testing it now, and it does push the query.

Could you shot a video of what's not working? or describe what are you doing?

**Reply by icyJoseph** [reply]
Did you fix it?
